lib/puppet/transaction/persistence.rb

```diff
@@ -0,0 +1,81 @@
+require 'yaml'
+require 'puppet/util/yaml'
+
+# A persistence store implementation for storing information between
+# transaction runs for the purposes of information inference (such
+# as calculating corrective_change).
+# @api private
+class Puppet::Transaction::Persistence
```

Comment:
Its cool, its nice to know I wasn't the only one to consider this path :-). But yeah, devil in the details.